File: 2099. Find Subsequence of Length K With the Largest Sum/maxSubsequence.js
/**
 * @param {number[]} nums
 * @param {number} k
 * @return {number[]}
 */
var maxSubsequence = function (nums, k) {
    let pairs = []
    let res = []

    for (let i = 0; i < nums.length; i++) {
        pairs.push({ "num": nums[i], "idx": i })
    }
    pairs.sort((a, b) => b.num - a.num)

    let sorted = pairs.slice(0, k)

    sorted.sort((a, b) => a.idx - b.idx)

    for (let { num, idx } of sorted) {
        res.push(num)
    }

    return res
};

File: 2099. Find Subsequence of Length K With the Largest Sum/maxSubsequence.cpp
#include <vector>
#include <algorithm>

using namespace std;

class Solution {
    public:
        vector<int> maxSubsequence(vector<int>& nums, int k) {
            vector<pair<int,int>> pairs;
            vector<pair<int,int>> sorted;
            int sz = nums.size();
            vector<int> res;
    
            for(int i=0 ; i<sz; i++){
                pairs.push_back({nums[i] , i});
            }
    
            sort(pairs.begin(), pairs.end());
    
            for(int i= sz-k; i<sz; i++){
                sorted.push_back(pairs[i]);
            }
    
            sort(sorted.begin(), sorted.end(), [](const pair<int,int>& a, const pair<int,int>& b){
                return a.second < b.second;
            });
    
            for(const auto& s : sorted){
                res.push_back(s.first);
            }
    
            return res;
        }
    };